builders/ethereum/precompiles/utility/non-specific.md

```diff
@@ -0,0 +1,91 @@
+keywords: ethereum, moonbeam, StorageCleaner, ECRecoverPublicKey
+---
+
+# Non-Network Specific Precompiled Smart Contracts
+
+## Introduction {: #introduction }
+
+A precompiled contract, or precompile, is a set of programmed functionalities hard-coded into the blockchain client. Precompiles perform computationally heavy tasks, such as cryptographic processes like hashing. Moving these functionalities to the blockchain client serves the dual purpose of making the computation more efficient than using a traditional smart contract and ensuring everyone has access to the complete and accurate set of processes and algorithms required to operate correctly.
+
+Precompile functionality is bundled and shared under a smart contract address which allows interactions similar to those of a traditional smart contract. Some precompiled contracts are not specific to Ethereum or to Moonbeam, but are supported for use in your Moonbeam application. 
+
+The nonspecific precompiles currently included in this category include `StorageCleaner` and `ECRecoverPublicKey`. 
+
+In the next section, you will learn more about the functionalities included in these precompiles.  
+
+## Clear Storage Entries with StorageCleaner {: #clear-storage-with-storagecleaner }
+
+The primary function of the `StorageCleaner` precompile is to clear storage entry key-value pairs for a smart contract marked as self-destructed, previously referred to as 'suicided.' `StorageCleaner` includes functionality to iterate over a list of addresses to identify self-destructed contracts and delete the appropriate storage entries associated with identified addresses. You can also input a numeric limit to prevent the precompile from consuming too much gas. 
+
+With the implementation of [EIP-6780: SELFDESTRUCT](https://eips.ethereum.org/EIPS/eip-6780) as part of the Ethereum Cancun/Dencun upgrade, contracts can only be self-destructed in the same transaction where they are created. This limitation keeps storage entries small and allows them to be automatically deleted during destruction. The `StorageCleaner` precompile remains available when a legacy contract needs storage entries cleared. 
+
+## Retrieve a Public Key with ECRecoverPublicKey {: verifying-signatures-ecrecoverpublickey }
+
+The primary function of the `ECRecoverPublicKey` precompile is to recover the public key used to create a digital signature from a given message hash and signature. This precompile is similar to [ECRecover](/builders/ethereum/precompiles/utility/eth-mainnet/#verify-signatures-with-ecrecover/){target=\_blank}, with the exception of returning the public key of the account that signed the message rather than the account address. 
+
+In the following sections, you will learn how to use the `ECRecoverPublicKey` precompile.
+
+### Checking Prerequisites {: checking-prerequisites }
+
+--8<-- 'text/_common/install-nodejs.md'
```

Comment:
<img width="1088" alt="Screenshot 2024-08-15 at 4 13 01 PM" src="https://github.com/user-attachments/assets/34b0fb96-c74e-4fd0-92c6-0c854b3ccd50">

It would be great to see this as a split tab like the Ubuntu Mac OS above rather than sequential, but not a huge deal either way